activerecord/lib/arel/nodes/sql_literal.rb

```diff
@@ -8,12 +8,53 @@ class SqlLiteral < String
       include Arel::AliasPredication
       include Arel::OrderPredications
 
+      attr_reader :bypass_numeric_quoting
+
+      def initialize(value, bypass_numeric_quoting: false)
+        super(value.to_s)
+        @bypass_numeric_quoting = bypass_numeric_quoting && is_number?(value)
```

Comment:
If you pass `bypass_numeric_quoting: true` and a non-numeric value, it will silently be `false`. I'm not sure that is ok but I think the check has to exist so the user does not pass `true` only to bypass the quoting.